prometheus/CMakeLists.txt

```diff
@@ -3,13 +3,19 @@ cmake_minimum_required(VERSION 2.8)
 project("prometheus-client")
 set(PROJECT_VERSION "0.99")
 set(PROJECT_DESCRIPTION "Prometheus C++ client library")
+set(CMAKE_MACOSX_RPATH true)
 
 include(GNUInstallDirs)
 include(FindPkgConfig)
+if (APPLE)
+  # Find homebrew's libicu on Mac OS although it is hidden pretty well
+  set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/opt/icu4c/lib/pkgconfig")
```

Comment:
Yes, hacky and elegant at the same time... ;)

Here is the story: Mac OS ships it's own version of libicu(core) but: They do not provide header files for this library which makes it impossible to use in your own applications. It still lurks in the public system library path, though. 

This forces every downstream package manager (homebrew, macports, fink, you name it) to do _something_ with their versions of libicu. On one hand people need it, because Apples version is not usable, on the other hand they can't just plomp it into the library search path without breaking things.

The homebrew solution is to install the whole libicu package into a non-default but conforming location: `/usr/local/opt/icu4c/` (which is how the whole PUBLIC/PRIVATE business popped up).

To help `pkg_check_modules(...)` to find the library I inject this somewhat specific path into the pkg-config search path. That way the package is detected cleanly without further tinkering. So, it now works out of the box with homebrew (the current defacto standard) while it is still possible to set environment variable manually.

So, yes it's a hack. But the _original_ hack is with Apple (not publishing headers). This merely is our counter-hack.  And I'm pretty sure it is the _right_ hack. ;)